cfme/tests/middleware/test_middleware.py

```diff
@@ -0,0 +1,21 @@
+import pytest
+from utils.browser import ensure_browser_open
+from cfme.login import login_admin
+from cfme.fixtures import pytest_selenium as sel
+
+
+@pytest.fixture
+def webDriver(request):
+    ensure_browser_open()
+    login_admin()
+
+    def closeSession():
+        """ Place Holder for after-test action """
+    request.addfinalizer(closeSession)
+
+    return
+
+
+def test_middleware():
+    sel.force_navigate('middleware')
```

Comment:
yes, the fixture is not needed here and please use `snake_case` everywhere in our python codebase instead of `camelCase`. Also when you write fixtures, it is good to state the scope explicitly, like

```python
@pytest.fixture(scope='module')
def somefixture(...):
    # ...
```

We also "abuse" the decorator functionality to do this:
```python

# do something
# blah blah

@request.addfinalizer
def _somecleanup():
    # do cleanup

# do something extremely dangerous
```

It exactly the same effect as defining the function and then passing it in the `addfinalizer`.

You can avoid most of addfinalizers in fixtures using yield fixtures:

```python
@pytest.yield_fixture(scope='module')
def somefixture():
    # set up
    yield something
    # tear down
```

The fixture runs until yield which specifies the value to be returned as the fixture value, then when the test in the scope ends, the execution continues after yield.

*I think this is worth putting in our guide too :)*